Assets/MixedRealityToolkit/Utilities/Editor/Preferences/ProjectPreferences.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Utilities.Editor
+{
+    /// <summary>
+    /// Utility to save preferences that should be saved per project (i.e to source control) across MRTK. Supports primitive preferences bool, int, and float
+    /// </summary>
+    public class ProjectPreferences : ScriptableObject
```

Comment:
I wonder if we can test that project preferences will work. For example, a test that checks:

1) If a preference does not exist, it gets added with default values.
2) If we modify a preference, the preference after reload gets saves.

If we cannot do this as part of an automated test, how about adding this as a step in build call? I'm thinking about 2)

Reply:
hmmm we should be able to do a test fine. ProjectPreferences is ignored in MRTK repo so even if a test somehow fails and doesn't clean up properly, it can't go in. It will still only impact locally